libafl/src/corpus/inmemory_ondisk.rs

```diff
@@ -444,13 +444,11 @@ impl<I> InMemoryOnDiskCorpus<I> {
 
         if self.locking {
             self.store_input_from(testcase)?;
-        } else {
-            if let Err(error) = self.store_input_from(testcase) {
-                log::error!(
-                    "An error occurred when trying to write a testcase without locking: {}",
-                    error
-                );
-            }
+        } else if let Err(error) = self.store_input_from(testcase) {
+            log::error!(
+                "An error occurred when trying to write a testcase without locking: {}",
```

Comment:
This is like a format string, right? So we can do
`"An error occurred when trying to write a testcase without locking: {err}",`

Reply:
Micro optimization, but the whole thing could be
```rust
if let Err(err) = self.store_input_from(testcase) {
   if self.locking {
       return Err(err);
   }
   log::error!("An error occurred when trying to write a testcase without locking: {err}",
}
```
then you only need to check the extra condition in the error case

Reply:
applied.